src/compiler/transformers/module/module.ts

```diff
@@ -88,11 +88,14 @@ namespace ts {
             addRange(statements, endLexicalEnvironment());
             addExportEqualsIfNeeded(statements, /*emitAsReturn*/ false);
 
+            if (!currentModuleInfo.exportEquals) {
+                append(statements, createUnderscoreUnderscoreESModule());

```

Comment:
@rbuckton  thank you for the thorough explanation! To be safe, we should then emit `__esModule` first. There seems to be safer than emit at the end